Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,302 @@
+        for (java.lang.reflect.Field field : fields) {
+
+            LuceneField searchFieldAnnotation = field.getAnnotation(LuceneField.class);
+            field.setAccessible(true);
+            if (searchFieldAnnotation != null) {
+
+                try {
+                    Object contentOfField = field.get(entry);
+                    if (contentOfField != null) {
+                        doc.add(new StringField(searchFieldAnnotation.lucenefieldName(),
+                                String.valueOf(contentOfField).replace(" ", ""), Field.Store.YES));
+                    }
+
+                } catch (IllegalArgumentException ie) {
+                    throw new IllegalArgumentException(ie);
+                } catch (IllegalAccessException iae) {
+                    throw new IllegalAccessException(iae.getMessage());
+                }
+            }
+        }
+
+        doc.add(new StoredField(LuceneFieldNames.TYPE_ID, entry.getTypeId() != null ? entry.getTypeId() : ""));
+        doc.add(new StoredField(LuceneFieldNames.CREATOR, entry.getCreatorId() != null ? entry.getCreatorId() : ""));
+        doc.add(new StoredField(LuceneFieldNames.MODIFIED, entry.getModified() != null ? entry.getModified() : ""));
+        Calendar cal = Calendar.getInstance();
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSSZ");
+        doc.add(new StoredField(LuceneFieldNames.MODIFIED_TIME, formatter.format(cal.getTime())));
+        doc.add(new StoredField(LuceneFieldNames.ID, entry.getId() != null ? entry.getId() : ""));
+
+        try {
+            writer.addDocument(doc);
+            writer.commit();
+        } catch (IOException ex) {
+            throw new LuceneException("Cannot insert concept in lucene. Please retry", ex);
+        }
+
+    }
+
+    private ConceptEntry getConceptFromDocument(Document d) throws IllegalAccessException {
+
+        ConceptEntry con = new ConceptEntry();
+        java.lang.reflect.Field[] fields = con.getClass().getDeclaredFields();
+        for (java.lang.reflect.Field field : fields) {
+            LuceneField luceneFieldAnnotation = field.getAnnotation(LuceneField.class);
+            field.setAccessible(true);
+            try {
+                if (luceneFieldAnnotation != null && d.get(luceneFieldAnnotation.lucenefieldName()) != null)
+                    field.set(con, d.get(luceneFieldAnnotation.lucenefieldName()));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(e);
+            } catch (IllegalAccessException e) {
+                throw new IllegalAccessException();
+            }
+        }
+        con.setId(d.get("id"));
+        return con;
+    }
+
+    @Override
+    public void deleteIndexes() throws LuceneException {
+        try {
+            Query q = new QueryParser("conceptType", whiteSpaceAnalyzer).parse("*:*");
+            writer.deleteDocuments(q);
+            writer.commit();
```

Comment:
hm, works for me just fine. Did you recreate the searcher as well? Otherwise, it'll try to use a closed reader.
But looking at it again, just use 
```
writer.deleteAll()
```